Assets/Scripts/Managers/PlayerStateManager.cs

```diff
@@ -21,32 +16,147 @@ public enum State
     LayTronWall
 };
 
+/// <summary>
+/// Represents all of the possible states a player can be in. Micro states, or
+/// states that transition to another state automatically are designated with the 
+/// "_micro" suffix. 
+/// NOTE: If you add a state, you should add it the state to the [PlayerStateManager.states]
+/// </summary>
+public enum State : byte
+{
+    // Initial state
+    StartupState = 0, 
```

Comment:
Why assign the numeric values explicitly here? (I honestly don't understand if that's intentional -- the compiler does this anyways)

Reply:
The reason I did this is because eventually we will have to version our serialization (so that if we make changes while the game is live, we can be backwards compatible). So if we are creating and destroying states, we should only ever use never-before-used numbers for the new states. It's probably not necessary but it just ensures no issues.